### ISSUE TITLE
Question regarding pvalues derived from phyper function

### ISSUE THREAD

**Issue by ivan-marroquin** [mirkosignorelli/csv#1]
Hi,

Many thanks for making available this package.

I gave a try with my data. The network has this size # 902076 edges and # 296180 nodes . After I generated a community detection using leiden algorithm and run your script to compute CSV. I noticed that the computed pvalue in either "test_between" (refer to snapshot named pvalues_blist_not_provided) or "test_within" (refer to snapshot named pvalues_blist_provided) is 0.0

I wonder if this result is due to the relative large numbers in "x, ss, K, N" (as observed in both screen shots). I think that these values are somehow beyond the support limit for the hypergeometric funtion (phyper in R).

Any suggestions?

Ivan

[computed_pvalues.zip](https://github.com/mirkosignorelli/csv/files/8969153/computed_pvalues.zip)

**Comment by mirkosignorelli** [comment]
Dear Ivan,
judging from your screenshots, I think that the computations of the p-values is correct: in your large network there seems to be a lot of evidence to reject the null hypotheses (both within and between groups), and this makes the p-values so small that they are numerically 0. 
This is quite apparent from the rather huge difference between nab and expected_nab in your screenshots: tests within the groups show evidence that there are many more links within the groups than expected by chance, and tests between groups show the opposite pattern (considerably less links than expected).
To sum it up: the computations seem correct, and you can conclude that there is a lot of evidence that your partition of nodes induces a clear community structure.

**Comment by ivan-marroquin** [comment]
Dear Mirko,

Many thanks for your prompt answer. The comments you provided are helpful.

Hope you don't mind, I have a following question about the derive weighted/unweighted CSV. For the graph that I am using, I get these results (for a significance level of 0.05):

"unweighted CSV" 0.0 
"weighted CSV" 0.0 
"pcorr" 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571
 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571 0.28571
 0.28571 0.28571 0.28571

I chase in the code the reason of these outputs. It seems to me that they are caused by the relative large values for "nab, ss, K, N" used to compute NEAT. For instance, if I use these inputs:
nab= 1183, ss= 440555, K= 314578, N= 1804152 (see first row in pvalues_blist_not_provided.png)

p1= phyper(x, m = K, n = N-K, k = ss, lower.tail = FALSE) -> 1
p2= phyper(x-1, m = K, n = N-K, k = ss, lower.tail = FALSE) -> 0

It looks to me that these relative large number are, perhaps, beyond the support domain of phyper function. And I get a result from the tail of the distribution. Unfortunately, such results causes a domino effect that cascades down to the estimation of weighted/unweighted CSV.

So, is it possible "nab, ss, K, N" according to the size of the graph? Or is it better to work with much lower confidence levels?

What would you suggest?

Kind regards,

Ivan

**Comment by ivan-marroquin** [comment]
Dear Miko, 

Thanks for your help. There was a mistake in my code, once I fixed it. I got 1.0 for weighted/unweigted CSV - which matches the observation between nab and expected nab values.

Kind regards,
Ivan
